### PR TITLE
statistics: add the refresher as a stats owner listener

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

In this PR, I added the refresher as a listener for the stats owner, to ensure that after a stats owner changes, the background process can stop correctly.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/statistics/handle/types/interfaces.go [hunk 1]
```diff
@@ -158,6 +158,12 @@ type StatsAnalyze interface {
 	// CheckAnalyzeVersion checks whether all the statistics versions of this table's columns and indexes are the same.
 	CheckAnalyzeVersion(tblInfo *model.TableInfo, physicalIDs []int64, version *int) bool
 
+	// OnBecomeOwner is used to handle the event when the current TiDB instance becomes the stats owner.
+	OnBecomeOwner()
```
Comment (lance6716) [inline pkg/statistics/handle/types/interfaces.go:162]: I think embedding the `Listener` interface is better

**Review by lance6716 — APPROVED** [review]
lgtm for non-stats part

**Review by D3Hunter — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

File: pkg/statistics/handle/types/interfaces.go (hunk 1 above)
Comment (Rustin170506) [reply pkg/statistics/handle/types/interfaces.go]: Updated.

**Pull request merged**